### PR TITLE
UI: Namespace area fixes

### PR DESCRIPTION
**Pull request by hashishaw**
This PR fixes a couple issues with the namespaces area of the UI:

**Goes to the wrong link if namespace has dots**
wrong link when namespace with dots

**Shows blue info banner from unexpected param**
image

### After
namespace-fixes

### CODE REVIEW

File: ui/app/adapters/namespace.js
Comment (Monkeychip) [inline ui/app/adapters/namespace.js:1]: thanks for turning into class based!

File: ui/app/templates/vault/cluster/access/namespaces/index.hbs [hunk 1]
```diff
@@ -35,7 +35,7 @@
           {{#let (concat this.currentNamespace (if this.currentNamespace "/") list.item.id) as |targetNamespace|}}
             {{#if (includes targetNamespace this.accessibleNamespaces)}}
               <li class="action">
-                <NamespaceLink @targetNamespace={{targetNamespace}} @class="is-block">
+                <NamespaceLink @targetNamespace={{targetNamespace}} @unparsed={{true}} @class="is-block">
```
Comment (Monkeychip) [inline ui/app/templates/vault/cluster/access/namespaces/index.hbs:38]: would you mind adding documentation to the NamespaceLink component? Add `unparsed` parma, etc.

**Review by Monkeychip — APPROVED** [review]
Looks good to me! One request, but non-blocking.

File: ui/app/templates/vault/cluster/access/namespaces/index.hbs (hunk 1 above)
Comment (hashishaw) [reply ui/app/templates/vault/cluster/access/namespaces/index.hbs:38]: I'll add a comment about unparsed, but there's a lot in there that I have no idea what's going on 😬

**Pull request merged**